met-web/src/components/engagement/form/AddSurveyBlockTabs.tsx

```diff
@@ -1,14 +1,47 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Box from '@mui/material/Box';
 import TabContext from '@mui/lab/TabContext';
 import RichTextEditor from './RichTextEditor';
 import { MetTab, MetTabList, MetTabPanel } from './StyledTabComponents';
 import { upcomingText, openText, closedText } from 'constants/submissionStatusText';
 import { SubmissionStatus } from 'constants/engagementStatus';
+import { EngagementStatusBlock } from '../../../models/engagementStatusBlock';
 
-const AddSurveyBlockTabs = () => {
+const AddSurveyBlockTabs = ({
+    handleChange = (statusBlock: EngagementStatusBlock[]) => {
+        /* empty default method  */
+    },
+}) => {
+    // set initial value to upcoming
     const [value, setValue] = React.useState(SubmissionStatus[SubmissionStatus.Upcoming]);
 
+    // array to pass updated content to engagement context
+    const surveyBlockContent = [
+        { survey_status: SubmissionStatus[SubmissionStatus.Upcoming], block_text: '' },
+        { survey_status: SubmissionStatus[SubmissionStatus.Open], block_text: '' },
+        { survey_status: SubmissionStatus[SubmissionStatus.Closed], block_text: '' },
+    ];
+    const [contentList, setcontentList] = useState(surveyBlockContent);
+
+    // capture changes in richdescription
+    const handleRichDescriptionChange = (newState: string) => {
+        const updatedSurveyBlockContent = contentList.map((item) => {
+            if (item.survey_status === value && newState) {
+                item.block_text = newState;
+            }
+            return item;
+        });
+        setcontentList(updatedSurveyBlockContent);
+    };
+
+    // exclude content having empty block text
+    useEffect(() => {
+        const filterContentList = contentList.filter(function (el) {
+            return el.block_text != '';
+        });
+        handleChange(filterContentList);
```

Comment:
Do we need to manage the state in two places, here and in the context file?

Reply:
changed as per review comments